### ISSUE TITLE
event_rate_limit option does not work as expected.

### ISSUE THREAD

**Issue by NM4** [rancher/rke#2012]
**RKE version:**
1.1.0
**Docker version: (`docker version`,`docker info` preferred)**

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
RancheOS
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware
**cluster.yml file:**

```
    event_rate_limit:
      enabled: true
      configuration:
        apiVersion: eventratelimit.admission.k8s.io/v1alpha1
        kind: Configuration
        limits:
        - type: Server
          qps: 6000
          burst: 30000
```

**Steps to Reproduce:**
Create an rke config file including the excerpt above and then create a cluster
**Results:**
Th kube-apiserver don't start  with the following error message :

> W0403 09:53:24.972851       1 helpers.go:98] ignoring invalid key policy/v1beta1/podsecuritypolicy, individual resource enablement/disablement is not supported in policy/v1beta1, and will prevent starting in future releases
Error: failed to initialize admission: couldn't init admission plugin "EventRateLimit": Object 'Kind' is missing in '{"limits":[{"burst":30000,"qps":6000,"type":"Server"}]}'
Usage:
  kube-apiserver [flags]

gz#16845

**Comment by igorcezar** [comment]
I have the same issue here, with RKE 1.1.0 and RancherOS 1.5.5.

As far as I can see, the problem is that RKE creates the file /etc/kubernetes/admission.yaml with the wrong configuration.
When you set a custom configuration in rkeConfig file like this:
```
 event_rate_limit:
      enabled: true
      configuration:
        apiVersion: eventratelimit.admission.k8s.io/v1alpha1
        kind: Configuration
        limits:
        - type: Server
          qps: 6000
          burst: 30000
```
RKE creates this admission.yaml:

```
apiVersion: apiserver.k8s.io/v1alpha1
kind: AdmissionConfiguration
plugins:
- configuration:
    limits:
    - burst: 30000
      qps: 6000
      type: Server
  name: EventRateLimit
  path: ""
```
As you can see, it misses the Kind and ApiVersion parameters.
But if you do not set a custom configuration
```
 event_rate_limit:
      enabled: true
```
then RKE creates a default admission.yaml, with the correct configuration:
```
apiVersion: apiserver.k8s.io/v1alpha1
kind: AdmissionConfiguration
plugins:
- configuration:
    apiVersion: eventratelimit.admission.k8s.io/v1alpha1
    kind: Configuration
    limits:
    - burst: 20000
      qps: 5000
      type: Server
  name: EventRateLimit
  path: ""
```

For now this can be a workaround, so that kube-apiserver can work properly without the need to disable EventRateLimit.
If you need to change burst or qps parameters, you could do it manually in the admission.yaml of each master node, but RKE will change it everytime you do an upgrade.

**Comment by teknologista** [comment]
Strange, this seems to be the case for Secrets Encryption too...

Using something like:

```
services:
  kube-api:
    secrets_encryption_config:
      enabled: true
      custom_config:
        apiVersion: apiserver.config.k8s.io/v1
        kind: EncryptionConfiguration
        resources:
        - resources:
          - secrets
          providers:
          - aescbc:
              keys:
              - name: k-fw5hn
                secret: RTczRjFDODMwQzAyMDVBREU4NDJBMUZFNDhCNzM5N0I=
          - identity: {}
```

In cluster.yaml and running roe through terraform outputs:

`Failed to parse cluster config: error decoding data: Object 'Kind' is missing in '{"resources":[{"providers":[{"aescbc":{"keys":[{"name":"key","secret":"`

**Comment by maxisam** [comment]
I have this issue too. The bot should keep this alive.

**Comment by jgericke** [comment]
I seem to have just hit this:

 ```yaml:
   event_rate_limit:
      enabled: true
      configuration:
        apiVersion: eventratelimit.admission.k8s.io/v1alpha1
        kind: Configuration
        limits:
        - type: Server
          qps: 6000
          burst: 30000
```

`FATA[0084] [controlPlane] Failed to bring up Control Plane: [Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [flatcar-02.jlab.local]: Get "https://localhost:6443/healthz": Unable to access the service on localhost:6443. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused), log: Error: failed to initialize admission: couldn't init admission plugin "EventRateLimit": Object 'Kind' is missing in '{"limits":[{"burst":30000,"qps":6000,"type":"Server"}]}']`

```
-> % rke version
INFO[0000] Running RKE version: v1.2.11          
```

**Comment by cspwizard** [comment]
Is there any estimate when this issue will be fixed?

**Comment by jrowinski3d** [comment]
I too am receiving this issue. Just testing with the default provided in the docs. Running
**RKE VERSION** -> v1.3.7

**Comment by bheemareddy** [comment]
Hi Team, I'm trying to test the Event Rate Limit in the device.  Any suggestion on how to test this with the event rate limit admission control plugin and without.

Thanks in advance.

**Comment by angusfz** [comment]
Same issue here
RKE version: `v1.3.11` with below rate limit config in `cluster.yml`
```yaml
services:
  kube-api:
    event_rate_limit:
      enabled: true
      configuration:
        apiVersion: eventratelimit.admission.k8s.io/v1alpha1
        kind: Configuration
        limits:
        - type: Server
          qps: 6000
          burst: 30000
```
then `rke up`
```
ERRO[0118] Failed to upgrade hosts: rancherserver1 with error [Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [10.152.65.230]: Get "https://localhost:6443/healthz": Unable to access the service on localhost:6443. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused), log: E0704 05:39:24.143658       
1 run.go:74] "command failed" err="failed to initialize admission: couldn't init admission plugin \"EventRateLimit\": Object 'Kind' is missing in '{\"limits\":[{\"burst\":30000,\"qps\":6000,\"type\":\"Server\"}]}'"] 
```

If we set `evet_rate_limit` without `configuration:`,  then `rek up`  can run without issue
```yaml
services:
  kube-api:
    event_rate_limit:
      enabled: true
```

**Comment by jiaqiluo** [comment]
reopen the issue, it was closed automatically by GitHub.

**Comment by jiaqiluo** [comment]
waiting for RKE rc

**Comment by jiaqiluo** [comment]
### Root cause
 
In the backend, the parsing for EventRateLimit is missing, so the applied configuration for EventRateLimit does not take effect. 
 
### What was fixed, or what changes have occurred

add the missing logic for parsing and applying the EventRateLimit  

### Areas or cases that should be tested
When the configuration is in the cluster.yaml, RKE should apply it to the cluster. 
 
When setting the following configuration, the cluster is provisioned successfully, and the error reported in the issue doesn't happen anymore.

```services:
  kube-api:
    event_rate_limit:
      enabled: true
      configuration:
        apiVersion: eventratelimit.admission.k8s.io/v1alpha1
        kind: Configuration
        limits:
          - type: Server
            qps: 6999
            burst: 29999
```

### What areas could experience regressions?
no 
  
### Are the repro steps accurate/minimal?
yes

**Comment by jiaqiluo** [comment]
can be validated in RKE CLI v1.4.3-rc1

**Comment by kinarashah** [comment]
Available to test with `v1.4.4-rc3`

**Comment by slickwarren** [comment]
tested with rke v1.4.4-rc3:
* event_rate_limit enabled without a `configuration` set -> RKE should run without a configuration set -- pass
* event_rate_limit [enabled with valid](https://github.com/rancher/rke/issues/2012#issuecomment-1349996270) `configuration` set -> RKE should use the specified configuration -- pass

note: manually setting `configuration: null` will no longer be a valid config, so using a config with this previous workaround will no longer work.
